samples/hello_fpga.c

```diff
@@ -106,6 +113,132 @@ void print_err(const char *s, fpga_result res)
 	fprintf(stderr, "Error %s: %s\n", s, fpgaErrStr(res));
 }
 
+/*
+ * Global configuration of bus, set during parse_args()
+ * */
+struct config{
+	struct target {
+		int bus;
+		int open_flags;
+	} target;
+}
+
+config = {
+	.target = {
+		.bus = -1,
+		.open_flags = 0
+	}
+};
+
+#define GETOPT_STRING "B:s"
+fpga_result parse_args(int argc, char *argv[])
+{
+	struct option longopts[] = {
+		{"bus",		required_argument, NULL, 'B'},
+		{"shared access", no_argument,     NULL, 's'}
+	};
+	
+	int getopt_ret;
+	int option_index;
+	char *endptr = NULL;
+
+	while (-1 != (getopt_ret = getopt_long(argc, argv, GETOPT_STRING,
+						longopts, &option_index))) {
+		const char *tmp_optarg = optarg;
+		/* Checks to see if optarg is null and if not it goes to value of optarg */
+		if ((optarg) && ('=' == *tmp_optarg)){
+			++tmp_optarg;
+		}
+
+		switch (getopt_ret){
+		case 'B': /* bus */
+			if (NULL == tmp_optarg)
+				break;
+			endptr = NULL;
+			config.target.bus = (int) strtoul(tmp_optarg, &endptr, 0);
+			if (endptr != tmp_optarg + strnlen(tmp_optarg, 100)) {
+				fprintf(stderr, "invalid bus: %s\n", tmp_optarg);
+				return FPGA_EXCEPTION;
+			}
+			break;
+		case 's':
+			config.target.open_flags |= FPGA_OPEN_SHARED;
+			break;
+		
+		default: /* invalid option */
+			fprintf(stderr, "Invalid cmdline option \n");
+			return FPGA_EXCEPTION;
+		}
+	}
+	
+	return FPGA_OK;
+}
+
+
+int find_fpga(fpga_guid interface_id, fpga_token *accelerator_token, uint32_t *num_matches_accelerators, uint32_t *num_matches_fpga)
+{
+	fpga_properties filter = NULL;
+	fpga_result res;
+
+	/* Get number of FPGAs in system */
+	res = fpgaGetProperties(NULL, &filter);
+	ON_ERR_GOTO(res, out, "creating properties object");
+
+	res = fpgaPropertiesSetObjectType(filter, FPGA_DEVICE);
+	ON_ERR_GOTO(res, out_destroy, "setting interface ID");
+
+	res = fpgaPropertiesSetObjectType(filter, FPGA_ACCELERATOR);
+	ON_ERR_GOTO(res, out_destroy, "setting object type");
+
+	res = fpgaPropertiesSetGUID(filter, interface_id);
+	ON_ERR_GOTO(res, out_destroy, "setting GUID");
+
+	if (-1 != config.target.bus) {
+		res = fpgaPropertiesSetBus(filter, config.target.bus);
+		ON_ERR_GOTO(res, out_destroy, "setting bus");
+	}
+
+	res = fpgaEnumerate(&filter, 1, accelerator_token, 1, num_matches_accelerators);
+	ON_ERR_GOTO(res, out_destroy, "enumerating accelerators");
+	
+	res = fpgaEnumerate(&filter, 1, accelerator_token, 1, num_matches_fpga);
+	ON_ERR_GOTO(res, out_destroy, "enumerating FPGAs");
+
+out_destroy:
+	res = fpgaDestroyProperties(&filter);
+	ON_ERR_GOTO(res, out, "destroying properties object");
+
+out:
+	return err_cnt;
+}
+
+	
+/* function to get the bus number when there are multiple buses */
+struct bus_info{
+	uint8_t bus;
+};
```

Comment:
Why is this a struct?  You should replace this struct with a uint8_t bus only, unless you suspect it will be expanded in the future (most likely not).

Reply:
I left it as a struct just in case we would like to print information about the device and function if needed in the future.